pkg/rollout/revision.go

```diff
@@ -40,13 +41,30 @@ func DetectStableRevisionName(svc *run.Service) string {
 
 // DetectCandidateRevisionName attempts to deduce what revision could be
 // considered a candidate.
-func DetectCandidateRevisionName(svc *run.Service, stable string) string {
+//
+// It also determines if the candidate is different from a previous rollout
+// process. Knowing if a candidate is a new one is useful since metrics cannot
+// be obtained on it (it has 0 traffic), so the rollout process should add some
+// initial traffic to the new revision.
```

Comment:
I don't see anything about what the bool is in godoc.

Reply:
We decided to take the logic to determine if candidate is new outside this function.